java/vector/src/main/java/org/apache/arrow/vector/complex/MapVector.java

```diff
@@ -121,9 +128,29 @@ public UnionMapReader getReader() {
    */
   @Override
   public <T extends ValueVector> AddOrGetResult<T> addOrGetVector(FieldType fieldType) {
-    AddOrGetResult<T> result = super.addOrGetVector(fieldType);
+
+    // TODO: can call super method once DATA_VECTOR_NAME is configurable
+    boolean created = false;
+    if (vector instanceof ZeroVector) {
+      vector = fieldType.createNewSingleVector("entries", allocator, callBack);
```

Comment:
cc @pravindra on the name change for MapVector struct field from "$data$" to "entries"